core/src/main/scala/org/apache/spark/rdd/RDD.scala

```diff
@@ -1539,6 +1539,13 @@ abstract class RDD[T: ClassTag](
     // NOTE: we use a global lock here due to complexities downstream with ensuring
     // children RDD partitions point to the correct parent partitions. In the future
     // we should revisit this consideration.
+    if (doCheckpointCalled) {
+      throw new SparkException("Because job has been executed on this RDD, checkpoint won't work")
+    }
+    if (storageLevel == StorageLevel.NONE) {
+      logWarning("Call checkpoint on unpersisted RDD, it will cause RDD recomputation" +
+        " when saving checkpoint files.")
+    }
```

Comment:
en... your advice is reasonable!
to sovle (a), I can move this warning into `doCheckpoint`, avoid the situation `RDD.persist` called later.
about problem (b), how about change this warning into `logInfo` ?